### PR TITLE
[SP-1102] - GET/v2/info/versions update and removement of deprecated endpoints 

### PR DESCRIPTION
**Pull request by deexidee**
### Purpose

Swagger doc description for endpoint "/v2/info/versions" at rest/impl/Version.class and removal of endpoints marked @Deprecated(since = "0.71.0", forRemoval = true).

### Breaking changes
Removal of deprecated API endpoints:
* `/v2/pipelines/own`
* `/v2/pipelines/system`
* `/v2/streams/own`
* `/v2/streams/own/<element_id>`

 ### Remarks

PR introduces (a) breaking change(s): yes

PR introduces (a) deprecation(s): no

### CODE REVIEW

**Review by tenthe — COMMENTED** [review]
Great, thank you very much!

**Comment by tenthe** [comment]
They failed due to some checkstyle problems, see: https://github.com/apache/streampipes/actions/runs/4053261489/jobs/6981510149

Here is described how you can set this up checkstyle locally: https://cwiki.apache.org/confluence/display/STREAMPIPES/Code+Style+-+Java

There is also an issue on this topic, see #880. This might be a good next topic for you if you are interested. When we add the validation as a git pre-commit hook, we will avoid such situations in the future.

File: streampipes-rest/src/main/java/org/apache/streampipes/rest/impl/Version.java [hunk 1]
```diff
@@ -33,6 +34,7 @@ public class Version extends AbstractAuthGuardedRestResource {
   @GET
   @Path("/versions")
```
Comment (bossenti) [inline streampipes-rest/src/main/java/org/apache/streampipes/rest/impl/Version.java:35]: I would propose to rename the endpoint to `check`, `health-check` or `status`.
Any other ideas?

File: streampipes-rest/src/main/java/org/apache/streampipes/rest/impl/Version.java [hunk 2]
```diff
@@ -33,6 +34,7 @@ public class Version extends AbstractAuthGuardedRestResource {
...
   @Produces(MediaType.APPLICATION_JSON)
+  @Operation(summary = "Provides health-check and information about current backend version.", tags = {"Version"})
```
[the remark this answers is not in the document]
Comment (bossenti) [reply streampipes-rest/src/main/java/org/apache/streampipes/rest/impl/Version.java:37]: Alternatively, we could provide this endpoint twice: with and without authentication

**Comment by bossenti** [comment]
Thanks a lot for your contribution @deexidee 🎉

File: streampipes-rest/src/main/java/org/apache/streampipes/rest/impl/Version.java (hunk 1 above)
Comment (bossenti) [reply streampipes-rest/src/main/java/org/apache/streampipes/rest/impl/Version.java:35]: @tenthe @dominikriemer

File: streampipes-rest/src/main/java/org/apache/streampipes/rest/impl/Version.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (dominikriemer) [reply streampipes-rest/src/main/java/org/apache/streampipes/rest/impl/Version.java:37]: We also have an endpoint in the `Setup` resource which is unauthenticated and is used  by the UI to determine whether the core is running upon startup. We could move the version info there and can keep the authenticated endpoint for more detailed information. What do you think?

File: streampipes-rest/src/main/java/org/apache/streampipes/rest/impl/Version.java (hunk 1 above)
Comment (dominikriemer) [reply streampipes-rest/src/main/java/org/apache/streampipes/rest/impl/Version.java:35]: We could rename this to Status and add more information on registered extension services and other status info.
What info do we want to return here?

**Review by bossenti — APPROVED** [review]
Thanks a lot for your contribution @deexidee 🙏🏼 
We would be glad if you are interested in further contributions :)

**Pull request merged**